src/test/ui/suggestions/issue-97677.stderr

```diff
@@ -8,8 +8,8 @@ LL |     n + 10
    |
 help: consider restricting type parameter `N`
    |
-LL | fn add_ten<N: std::ops::Add<i32>>(n: N) -> N {
-   |             ++++++++++++++++++++
+LL | fn add_ten<N: std::ops::Add<i32><Output=N>>(n: N) -> N {
+   |             ++++++++++++++++++++++++++++++
```

Comment:
I figured out the issue. I added a fix, although it duplicates some code as you can see in the commit and the FIXME. If you want me to figure out a way to factor out the duplicated code I can do that, otherwise the PR is ready I think.